### DISCUSSION TITLE
What is the approach being used for voice cloning?

### DISCUSSION THREAD

**Discussion by charansingh9** [coqui-ai/TTS#3285] [General Q&A]
I was going through the repo and using the inference functions to get the inference for voice cloning using fairseq models. I am curious to know what is the approach being used for the voice cloning as I have also seen voice conversion modules in the repo.

In the TTS-Paper repo I have found this paper being used for voice conversion:
https://arxiv.org/pdf/2210.15418.pdf

But this paper takes target and source audios, and convert the source audio to target audio voice. Will someone help me understand what is happening when we use

```
from TTS.api import TTS
api = TTS(model_name="tts_models/eng/fairseq/vits").to("cuda")
api.tts_to_file("This is a test.", file_path="output.wav")

api = TTS("tts_models/deu/fairseq/vits")
api.tts_with_vc_to_file(
    "Wie sage ich auf Italienisch, dass ich dich liebe?",
    speaker_wav="target/speaker.wav",
    file_path="ouptut.wav"
)
```

And what is happening when we use this for voice cloning:

```
tts = TTS("tts_models/multilingual/multi-dataset/xtts_v2").to(device)

wav = tts.tts(text="Hello world!", speaker_wav="my/cloning/audio.wav", language="en")

tts.tts_to_file(text="Hello world!", speaker_wav="my/cloning/audio.wav", language="en", file_path="output.wav")

```

As both of them are claimed to help with voice cloning, what is the difference? It would be so helpful if anyone could lead me to a resource to understand what is happening when we pass **text** and **reference audio** as input and the output is cloned voice.

Thanks in advance.

**Comment by eginhard** [comment]
```python
TTS("tts_models/deu/fairseq/vits").tts_with_vc_to_file(
    "Wie sage ich auf Italienisch, dass ich dich liebe?",
    speaker_wav="target/speaker.wav",
    file_path="output.wav"
)
```
This does TTS with the model you specified and then does voice conversion with a separate FreeVC model, i.e. it is exactly the same as:
```python
TTS("tts_models/deu/fairseq/vits").tts_to_file(
    "Wie sage ich auf Italienisch, dass ich dich liebe?",
    file_path="intermediate.wav"
)
TTS("voice_conversion_models/multilingual/vctk/freevc24").voice_conversion_to_file(
    source_wav="intermediate.wav",
    speaker_wav="target/speaker.wav",
    file_path="output.wav"
)
```

XTTS has integrated voice cloning, i.e. it can directly generate speech in the target speaker's voice without a separate VC model.

**Reply by charansingh9** [reply]
Thank you so much for the clarification. @eginhard
